lib/ruby_indexer/lib/ruby_indexer/entry.rb

```diff
@@ -69,6 +78,23 @@ class Class < Namespace
       def initialize(name, file_path, location, comments, parent_class)
         super(name, file_path, location, comments)
         @parent_class = T.let(parent_class, T.nilable(String))
+        @singleton_klass = T.let(SingletonClass.new(name, self), SingletonClass)
```

Comment:
Isn't it already done in the `Namespace` parent?

Reply:
Yup, fixed.